lighthouse-cli/test/smokehouse/lighthouse-runners/bundle.js

```diff
@@ -66,14 +94,65 @@ async function runLighthouse(url, configJson, testRunnerOptions = {}) {
     return {
       lhr: runnerResult.lhr,
       artifacts: runnerResult.artifacts,
-      log: '', // TODO: if want to run in parallel, need to capture lighthouse-logger output.
     };
   } finally {
     // Clean up and return results.
     await launchedChrome.kill();
   }
 }
 
+/**
+ * Launch Chrome and do a full Lighthouse run via the Lighthouse DevTools bundle.
+ * @param {string} url
+ * @param {LH.Config.Json=} configJson
+ * @param {{isDebug?: boolean, useFraggleRock?: boolean}=} testRunnerOptions
+ * @return {Promise<{lhr: LH.Result, artifacts: LH.Artifacts, log: string}>}
+ */
+async function runLighthouse(url, configJson, testRunnerOptions = {}) {
+  /** @type {string[]} */
+  const logs = [];
+  const workerFilename = `${LH_ROOT}/lighthouse-cli/test/smokehouse/lighthouse-runners/bundle.js`;
+  const worker = new Worker(workerFilename, {
+    stdout: true,
+    stderr: true,
+  });
+  worker.stdout.setEncoding('utf8');
+  worker.stderr.setEncoding('utf8');
+  worker.stdout.addListener('data', (data) => {
+    process.stdout.write(data);
+    logs.push(`STDOUT: ${data}`);
+  });
+  worker.stderr.addListener('data', (data) => {
+    process.stderr.write(data);
+    logs.push(`STDERR: ${data}`);
+  });
+  const promise = new Promise((resolve, reject) => {
+    worker.on('message', (message) => {
+      if (message.type === 'result') {
+        resolve(message.value);
+      } else if (message.type === 'error') {
+        reject(message.value);
+      }
+    });
+  });
+  worker.postMessage({url, configJson, testRunnerOptions});
+
+  const result = await promise;
```

Comment:
if this throws (any error passed back from the worker), then all the logs are lost. See the `ChildProcessError` used in the cli runner if that's helpful: https://github.com/GoogleChrome/lighthouse/blob/9039dbba4807793db76eca71a3bbb25583e10d12/lighthouse-cli/test/smokehouse/lighthouse-runners/cli.js#L120-L121